### PR TITLE
NetSim: Expire shard

### PR DESCRIPTION
**Pull request by islemaster**
Makes NetSim shards expire and completely reset after 2 hours of idle time.

Idle time is defined as two hours without an insert, update or delete operation.  We don't include the delete_all (manual shard reset) operation because it deletes everything anyway.

The expiration is implemented using [Redis' native EXPIRE command](redis.io/commands/expire).  For testing, I extended our FakeRedisClient to implement this method and to have a `time_travel` method for quickly simulating expiration.  The tests also work against real Redis, but take a bit longer because they must sleep until the real expiration occurs.

You can override the expire time in your development environment by adding the following line to your locals.yml:

```
netsim_shard_expire_time: 60
```

The value is measured in seconds.  A short expiration time (like a minute, above) may actually be useful for local development.

### CODE REVIEW

**Comment by islemaster** [comment]
The last commit uses [Redis' MULTI command](http://redis.io/commands/multi) to batch commands, so that adding EXPIRE commands does not increase the number of requests to Redis and have an adverse impact on performance (however minor).

The messiest part of this change was writing our own implementation of MULTI on the fake Redis used in tests; it has a fairly complex behavior where its return value is an array of the return values of all the redis commands executed in the block.

File: deployment.rb [hunk 1]
```diff
@@ -82,7 +82,7 @@ def load_configuration()
     'assets_s3_directory'         => rack_env == :production ? 'assets' : "assets_#{rack_env}",
     'sources_s3_bucket'           => 'cdo-v3-sources',
     'sources_s3_directory'        => rack_env == :production ? 'sources' : "sources_#{rack_env}",
-    'netsim_api_publickey'        => [:development].include?(rack_env) ? 'JGW2rHUp_UCMW_fQmRf6iQ==' : 'HQJ8GCCMGP7Yh8MrtDusIA==',
+    'netsim_shard_expire_time'    => 7200,
```
Comment (islemaster) [inline deployment.rb]: I'm also removing `netsim_api_publickey` because it's unused now, an artifact of being built on DynamoDB and the Channels API.

**Comment by islemaster** [comment]
@philbogle I think you are the best person to review this.

File: deployment.rb (hunk 1 above)
Comment (philbogle) [inline deployment.rb]: How about calling this netsim_shard_expiry_seconds  (to avoid someone getting the units wrong)

**Comment by philbogle** [comment]
LGTM

File: deployment.rb (hunk 1 above)
Comment (islemaster) [inline deployment.rb]: Sounds good!

**Pull request merged**